File: programmers/Level1_1_3.py
#체육복
def solution(n, lost, reserve):
    
    temp_list = []
    
    for i in reserve:
        if (i) in lost:
            print (i)
            temp_list.append(lost.pop(lost.index(i)))
            
    for i in temp_list:
        reserve.pop(reserve.index(i))
        
    for i in reserve:
        if (i-1) in lost:
            lost.pop(lost.index(i-1))
        elif (i+1) in lost:
            lost.pop(lost.index(i+1))
    
    answer = (n - len(lost))
    return answer

File: programmers/scala/Level1_1_3.scala
import scala.collection.mutable.ListBuffer

object Solution {
    def solution(n: Int, lost: Vector[Int], reserve: Vector[Int]): Int = {
    
        var temp_list:ListBuffer[Int] = ListBuffer()

        for (i <- reserve) {
          if (lost.contains(i)) {
            temp_list = temp_list += lost(i)
            lost.filterNot(_ == i)
            println(lost)
          }
        }
        for (i <- temp_list)
          reserve.filter(_ == i)
        for (i <- reserve) {
          if (reserve.contains(i-1))
            lost.filterNot(_ == i-1)
          else if (reserve.contains(i+1))
            lost.filterNot(_ == i+1)

        }
        print(lost)
        return (n-lost.length)
    }
}